### PR TITLE
xbps-src: make changes suggested by shellharden

### PR DESCRIPTION
**Pull request by ishaanbhimwal**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by classabbyamp** [comment]
most of these changes either do nothing for functionality or are outright wrong

File: xbps-src [hunk 1]
```diff
@@ -237,18 +237,18 @@ check_reqhost_utils() {
 
     [ "$IN_CHROOT" ] && return 0
 
-    for f in ${REQHOST_UTILS}; do
-        if ! command -v ${f} &>/dev/null; then
-            echo "${f} is missing in your system, can't continue!" 1>&2
+    for f in "$REQHOST_UTILS"; do
```
Comment (classabbyamp) [inline xbps-src:240]: this, for example, will run the loop only once with the entire value of REQHOST_UTILS, not once per whitespace-separated item in the value

**Comment by classabbyamp** [comment]
> just implement the correct bash syntax

`-z` in test is perfectly fine bash and POSIX sh syntax, it is not less correct than `= ""`

File: xbps-src [hunk 2]
```diff
@@ -325,7 +325,7 @@ setup_distfiles_mirror() {
     local mirror scheme path
 
     # Scheme file:// mirror locations only work with uchroot
-    for mirror in $XBPS_DISTFILES_MIRROR; do
+    for mirror in "${XBPS_DISTFILES_MIRROR[@]}"; do
```
Comment (Duncaen) [inline xbps-src:328]: This is not an array.

**Pull request closed without merging**